.github/workflows/pipeline.yml

```diff
@@ -10,10 +10,33 @@ on:
       - '**.md'
 
 jobs:
-  build:
-    name: Build and test
+  test:
+    name: Test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - run: docker build --target test --tag todo-app:test .
-    - run: docker run --name tasko-test todo-app:test
\ No newline at end of file
+    - run: docker run --name tasko-test todo-app:test
+
+  docker-build:
+    name: Build production docker image
+    needs: test
+    if: github.event.pull_request.merged == true && github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      - uses: actions/checkout@v3
+
+      - name: Build docker image
+        run: docker build --platform linux/amd64 --target production --tag alechasoftwire/todo-app:prod .
+
+      - name: Deploy to Docker Hub
+        run: docker push alechasoftwire/todo-app:prod
+
+      - name: Trigger Azure restart
+        run: curl -dH -X POST '${{ secrets.AZURE_WEBHOOK }}'
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400) which can lead to confusing pipeline results